lib/rucio/common/stomp_utils.py

```diff
@@ -15,145 +15,415 @@
-            brokers: "Sequence[str]",
-            port: int,
-            use_ssl: bool,
-            vhost,
-            reconnect_attempts: int,
-            ssl_key_file,
-            ssl_cert_file,
-            timeout,
-            heartbeats=(0, 1000),
-            *,
-            logger: "LoggerFunction" = logging.log
-    ) -> tuple[list, list]:
-
-        configuration_changed = any([
-            self._brokers != brokers,
-            self._port != port,
-            self._use_ssl != use_ssl,
-            self._vhost != vhost,
-            self._reconnect_attempts != reconnect_attempts,
-            self._ssl_key_file != ssl_key_file,
-            self._timeout != timeout,
-            self._heartbeats != heartbeats,
-        ])
-        if configuration_changed:
-            self._brokers = brokers
-            self._port = port
-            self._use_ssl = use_ssl
-            self._vhost = vhost
-            self._reconnect_attempts = reconnect_attempts
-            self._ssl_key_file = ssl_key_file
-            self._timeout = timeout
-            self._heartbeats = heartbeats
-
-        current_remotes = set(self._connections)
-        desired_remotes = set((ip, port) for ip in resolve_ips(brokers, logger=logger))
-
-        if configuration_changed:
-            # Re-create all connections
-            to_delete = current_remotes
-            to_create = desired_remotes
-        else:
-            to_delete = current_remotes.difference(desired_remotes)
-            to_create = desired_remotes.difference(current_remotes)
-
-            for remote in current_remotes.intersection(desired_remotes):
-                conn = self._connections[remote]
-
-                if self.is_stalled(conn, logger=logger):
-                    # Re-create stalled connections
-                    to_delete.add(remote)
-                    to_create.add(remote)
-
-        deleted_conns = []
-        for remote in to_delete:
-            conn = self._connections.pop(remote)
-            if conn.is_connected():
+                try:
+                    conn.connect(**params)
+                except ConnectFailedException as error:
+                    self._logger(logging.WARNING, "[broker] Could not deliver message due to "
+                                                  "ConnectFailedException: %s", str(error))
+                    continue
+                except Exception as error:
+                    self._logger(logging.ERROR, "[broker] Could not connect: %s", str(error))
+                    continue
+            try:
+                yield conn
+            except Exception:
+                self._logger(logging.ERROR, "[broker] Error in yielded code, skipping to next connection.")
+
+    def deliver_messages(self, messages: dict) -> list[int]:
+        """
+        Deliver messages.
+
+        Args:
+            messages (dict): Messages to deliver.
+
+        Returns:
+            list[int]: delivered message ids, ready for deletion.
```

Comment:
Please avoid specifying the argument types in the docstring if they are already present as type annotations; there's no need to duplicate this information and it can cause mismatches between the type annotation and the type specified in the docstring if changes are made down the line